clang/include/clang/Basic/Attr.td

```diff
@@ -1197,6 +1197,18 @@ def SYCLDevice : InheritableAttr {
   let Documentation = [SYCLDeviceDocs];
 }
 
+def SYCLGlobalVarSubject : SubsetSubject<Var, [{S->hasGlobalStorage() &&
+                                                !S->isLocalVarDecl()}],
+                                         "global variables">;
+
+def SYCLGlobalVar : InheritableAttr {
+  let Spellings = [GNU<"sycl_global_var">];
+  let Subjects = SubjectList<[SYCLGlobalVarSubject], ErrorDiag>;
+  let LangOpts = [SYCLIsDevice];
+  let Documentation = [SYCLGlobalVarDocs];
+  let SimpleHandler = 1;
```

Comment:
Rather than mark this with a simple handler, I think we now have a reason to write the handler in SemaDeclAttr.cpp -- that's where you can check the location of the attribute to ensure that it's written in a system header. Note, you will want to look at the expansion location of the attribute so that you can be sure uses from macros work properly. e.g.,
```
// sycl.hpp (or any system header)
#define FROBBLE __attribute__((sycl_global_var))
FROBBLE int i; // OK
__attribute__((sycl_global_var)) int j; // OK

// user.cpp (or any user file)
FROBBLE int k; // error
__attribute__((sycl_global_var)) int l; // error
```
Note that for the test cases, you can use line control markers to simulate a system header. e.g.,
```
// syntax is: # <line number> <string-literal> <flag>
# 1 "header.hpp" 1 // Simulate a #include to enter new file named header.hpp at line 1
# 1 "header.hpp" 3 // This header is a system header
// Put your system header declarations here
# N "header.hpp" 2 // Return from the simulated #include (replace N with the last line number of the "header.hpp" file)

// Put your user code here
```

Reply:
Latest commit addresses these two points.